plugins/amdgpu/amdgpu_plugin_topology.c

```diff
@@ -703,3 +832,347 @@ int topology_parse(struct tp_system *sys, const char *message)
 	topology_free(sys);
 	return ret;
 }
+
+static bool device_properties_match(struct tp_node *src, struct tp_node *dest)
+{
+	if (src->simd_count == dest->simd_count &&
+	    src->mem_banks_count == dest->mem_banks_count &&
+	    src->io_links_count == dest->io_links_count &&
+	    src->max_waves_per_simd == dest->max_waves_per_simd &&
+	    src->lds_size_in_kb == dest->lds_size_in_kb &&
+	    src->wave_front_size == dest->wave_front_size &&
+	    src->array_count == dest->array_count &&
+	    src->simd_arrays_per_engine == dest->simd_arrays_per_engine &&
+	    src->cu_per_simd_array == dest->cu_per_simd_array &&
+	    src->simd_per_cu == dest->simd_per_cu &&
+	    src->max_slots_scratch_cu == dest->max_slots_scratch_cu &&
+	    src->vendor_id == dest->vendor_id &&
+	    src->device_id == dest->device_id &&
+	    src->num_sdma_engines == dest->num_sdma_engines &&
+	    src->num_sdma_xgmi_engines == dest->num_sdma_xgmi_engines &&
+	    src->num_sdma_queues_per_engine == dest->num_sdma_queues_per_engine &&
+	    src->num_cp_queues == dest->num_cp_queues &&
+	    src->capability == dest->capability &&
+	    src->vram_public == dest->vram_public &&
+	    src->vram_size <= dest->vram_size &&
+	    src->num_gws <= dest->num_gws &&
+	    src->caches_count <= dest->caches_count &&
+	    src->fw_version <= dest->fw_version &&
+	    src->sdma_fw_version <= dest->sdma_fw_version) {
+		return true;
+	}
+	return false;
+
+}
+
+static bool iolink_match(struct tp_iolink *src, struct tp_iolink *dest)
+{
+	if (!src->valid)
+		return true;
+
+	if (!dest->valid)
```

Comment:
If !dest->valid, I think that should not be considered a match.

Reply:
ACK